internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -349,3 +369,22 @@ func dryRunPatchHelperFactory(c client.Client) structuredmerge.PatchHelperFactor
 		return structuredmerge.NewTwoWaysPatchHelper(original, modified, c, opts...)
 	}
 }
+
+func (r *Reconciler) requestIsRateLimited(ctx context.Context, req reconcile.Request) (bool, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// If no rateLimitingCache is defined return false.
+	if r.rateLimitingCache == nil {
+		return false, nil
+	}
+	if _, found, err := r.rateLimitingCache.Get(req.String()); found && err == nil {
```

Comment:
```suggestion
   // return true if the object is rate limited (the object has been processed since now - TTL)
	if _, found, err := r.rateLimitingCache.Get(req.String()); found && err == nil {
```